packages/react-ui/src/app/connections/connection-table.tsx

```diff
@@ -186,14 +75,180 @@ const filters = [
 function AppConnectionsTable() {
   const [refresh, setRefresh] = useState(0);
   const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [selectedRowIds, setSelectedRowIds] = useState<
+    Array<AppConnectionWithoutSensitiveData>
+  >([]);
   const { checkAccess } = useAuthorization();
   const { toast } = useToast();
-  const [searchParams] = useSearchParams();
+
+  const columns: ColumnDef<
+    RowDataWithActions<AppConnectionWithoutSensitiveData>,
+    unknown
+  >[] = [
+    {
+      id: 'select',
+      header: ({ table }) => (
+        <Checkbox
+          checked={
+            table.getIsAllPageRowsSelected() ||
+            (table.getIsSomePageRowsSelected() && 'indeterminate')
```

Comment:
what is this condition ? I see it everywhere, like why is this string 'indeterminate' in the condition ?